presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -197,7 +197,20 @@ default int compareTo(int leftPosition, int leftOffset, int leftLength, Block ri
      * <p>
      * The returned block must be a compact representation of the original block.
      */
-    Block copyPositions(List<Integer> positions);
+    Block copyPositions(int[] positions, int offset, int length);
+
+    /**
+     * Returns a block containing the specified positions.
+     * All specified positions must be valid for this block.
+     * Use primitive version of this method if possible
+     * since it allows to avoid integer boxing.
+     * <p>
+     * The returned block must be a compact representation of the original block.
+     */
+    default Block copyPositions(List<Integer> positions)
```

Comment:
Did you consider marking this method as deprecated as it is not supposed to be used.

Reply:
I'm concerned that anyone who uses this method will now get a little slower (due to the mapToInt and materializing an int[]). We might want to update all usages of this method and get rid of it as part of this PR.

Reply:
I have removed non-test usages of `copyPositions(List<Integer> ...)`

Reply:
There are not that many usages in tests. It's better to do them all and completely remove this method from API.